kmod/patch/Makefile

```diff
@@ -18,12 +18,15 @@ $(KPATCH_NAME)-objs += patch-hook.o output.o
 
 all: $(KPATCH_NAME).ko
 
-$(KPATCH_NAME).ko:
+build.env:
+	export -p > $@
```

Comment:
I also often find myself in need of *_SIZE variables when working on create-diff-object. Maybe we can move this to kpatch-build instead so it can be used in both scenarios?

Reply:
That's a good suggestion, I've had to manually set those up for `create-diff-object` in the past as well.

As for implementing that, `create-diff-object` is called before `KCFLAGS` and `KCPPFLAGS` are exported.  I think we can move those exports to before `create-diff-object` if needed.

Also, the Makefile approach dumps `build.env` into the livepatch-patch directory.  If we move `build.env` creation into `kpatch-build` do we drop it into the same location?

Reply:
Could it go in ~/.kpatch/tmp?  If the user runs with '-d' or --skip-cleanup it would automatically be where they need it.

Reply:
> As for implementing that, create-diff-object is called before KCFLAGS and KCPPFLAGS are exported. I think we can move those exports to before create-diff-object if needed.

Agree, we should re-arrange capture them as well. While we plan to capture the build environment, let's capture all the exported variables.

> Also, the Makefile approach dumps build.env into the livepatch-patch directory. If we move build.env creation into kpatch-build do we drop it into the same location?

Like @jpoimboe suggested, `~/.kpatch/tmp/` sounds like a nature fit to place the `build.env` file.

Reply:
Is it worth adding a `do_export()` wrapper function that does the export and also runs the `export -p > "$TEMPDIR"/build.env`.  This way the exports can stay where they are (close to related code) and maybe it's useful if other parts of the script happen to fail.

Also `kpatch-build.env` a better filename?

Reply:
> Is it worth adding a do_export() wrapper function that does the export and also runs the export -p > "$TEMPDIR"/build.env. This way the exports can stay where they are (close to related code) and maybe it's useful if other parts of the script happen to fail.

That'll be a lot of overwrites but makes it much easier, we also won't have to make sure new exports are added before dumping them. So it is probably worth it.

> Also kpatch-build.env a better filename?

Makes it clearer, yeah.